tests/shutdown/parent.c

```diff
@@ -120,6 +122,114 @@ int test4(const char* test_name)
     return 0;
 }
 
+// main parent waits for spawned child to shutdown before exiting
+// Child has no SIGGHUP handler
+// Child throws an assert on a child thread
+
+static void* _test5_thread_assert(void* arg)
+{
+    printf("hello from asserting thread\n");
+    assert(0);
+    return NULL;
+}
+
+int test5(const char* test_name)
+{
+    int r;
+    pid_t pid = 0;
+    char* child_argv[] = {"/bin/child", (char*)test_name, NULL};
+    int wstatus;
+
+    r = fork();
+    if (r < 0)
+    {
+        printf("r=%d\n", r);
+        assert("failed to fork child" == NULL);
+    }
+    else if (r == 0)
+    {
+        pthread_t thread;
+        int r;
+
+        printf("=== start test (%s)\n", __FUNCTION__);
+
+        if ((r = pthread_create(&thread, NULL, _test5_thread_assert, NULL)))
+        {
+            printf("pthread_create() failed: %d", r);
+            abort();
+        }
+
+        if (pthread_join(thread, NULL) != 0)
+        {
+            printf("pthread_join() failed");
+            abort();
+        }
+    }
+    else
+    {
+        int wstate = 0;
+        r = waitpid(r, &wstate, 0);
+        assert(r > 0);
+        assert(WTERMSIG(wstate));
+        assert(WTERMSIG(wstate) == SIGABRT);
+    }
+    return 0;
+}
+
+// main parent waits for spawned child to shutdown before exiting
+// Child has no SIGGHUP handler
+// Child throws an SEGV on a child thread
+static void* _test6_thread_assert(void* arg)
+{
+    printf("hello from crashing thread\n");
+    char* crash = NULL;
+    crash[0] = 'a';
+    return NULL;
+}
+
+int test6(const char* test_name)
```

Comment:
test5 and test6 can possibly share the thread creation and status checking bits.

Reply:
I like to keep them separate unless the code is really complex. I have seen a situation where common code for a test is changed and one of the tests then does not test what it is supposed to